src/main/kotlin/org/rust/ide/annotator/fixes/ConvertLetDeclTypeFix.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.presentation.renderInsertionSafe
+import org.rust.lang.core.psi.RsLetDecl
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.types.ty.Ty
+
+/**
+ * Change the declared type of a local variable.
+ */
+class ConvertLetDeclTypeFix(
+    decl: RsLetDecl,
+    private val fixText: String,
+    private val ty: Ty
+) : LocalQuickFixAndIntentionActionOnPsiElement(decl) {
+    override fun getFamilyName(): String = "Convert type of local variable"
+    override fun getText(): String = fixText
+
+    override fun invoke(
```

Comment:
I think it shouldn't be available for `let x: wrong_type = y` if `Specify type explicitly` intention can't be applied for `let x = y`. For example:

![image](https://user-images.githubusercontent.com/6079006/86308916-07dca780-bc23-11ea-97c6-c2fa575ecee4.png)

`let y: &impl T` will not compile.

Reply:
Thanks. I added check for `TyAnon` and changed the text to refer only to the name of the variable and not to the binding mode (`mut`/`ref`).